### DISCUSSION TITLE
How to use for Markdown validation?

### DISCUSSION THREAD

**Discussion by meglio** [thephpleague/commonmark#652] [Q&A] [answered]
Please provide an example of how would I use this library to validate Markdown and get a boolean TRUE/FALSE as a result of validation?

**Comment by colinodell** [comment, marked as answer]
According to [section 2.1 of the CommonMark spec](https://spec.commonmark.org/0.28/#characters-and-lines) any plain-text file is valid:

> Any sequence of characters is a valid CommonMark document.
>
> A character is a Unicode code point. Although some code points (for example, combining accents) do not correspond to characters in an intuitive sense, all code points count as characters for purposes of this spec.
>
> This spec does not specify an encoding; it thinks of lines as composed of characters rather than bytes. A conforming parser may be limited to a certain encoding.

We don't perform this validation in our library, but you could use something like [this](https://stackoverflow.com/questions/3670593/how-can-i-tell-if-a-file-is-text-using-php) to check the MIME type and return `TRUE` for plain-text documents.
